File: Spring/Chapter4/java_demo/demo/src/main/java/com/aspectj_annotation/AnnotationImpl.java
package com.aspectj_annotation;

import org.springframework.stereotype.Component;

@Component
public class AnnotationImpl implements AnnotationInterface {
    @Override
    public void save() {
        System.out.println("保存");
    }

    @Override
    public void modify() {
        System.out.println("修改");
    }

    @Override
    public void delete() {
        System.out.println("删除");
//        注释下面这行语句开启异常通知
//        int a = 1/0;
    }
}

File: Spring/Chapter4/kotlin_demo/demo/src/main/kotlin/aspectj_annotation/AnnotationImpl.kt
package aspectj_annotation

import org.springframework.stereotype.Component

@Component
class AnnotationImpl : AnnotationInterface{
    override fun save() {
        println("保存")
    }

    override fun modify() {
        println("修改")
    }

    override fun delete() {
        println("删除")
//        异常通知
//        val a = 1/0
    }
}